### PR TITLE
Feature/ccb dom element option

### PR DESCRIPTION
**Pull request by Pickra**
Please review @kgroat 

## New DOM element example
screen shot 2018-12-14 at 1 38 21 pm

## New code file example
screen shot 2018-12-14 at 1 38 08 pm

## New code string example
screen shot 2018-12-14 at 1 38 16 pm

### CODE REVIEW

File: examples/helpers.js [hunk 1]
```diff
@@ -9,10 +9,13 @@ const displayOpts = opts => '{' + Object.entries(opts).map(arr => {
     return ` ${key}: '${val}'`;
 }) + '}';
 
-export const usageExample = (opts = null) => {
+export const usageExample = (opts = null, codeFileExt = '') => {
     const options = !opts ? '' : `const options = ${displayOpts(opts)};`;
     const optionsExample = !opts ? '' : ', options';
 
+    const codeFile = !!codeFileExt ? `import * as displayCode from './path/fileName.${codeFileExt}'` : '';
```
Comment (sahiltalwar88) [inline examples/helpers.js]: might be easier to read as 
```codeFileExt && `import * as displayCode from './path/fileName.${codeFileExt}'` || ''```

File: examples/helpers.js [hunk 2]
```diff
@@ -9,10 +9,13 @@ const displayOpts = opts => '{' + Object.entries(opts).map(arr => {
...
+    const codeString = !codeFileExt ? "const displayCode = 'Just pretend this is the Example Code from above';" : '';
```
Comment (sahiltalwar88) [inline examples/helpers.js]: similarly,might be easier to read as 
```codeFileExt || "const displayCode = 'Just pretend this is the Example Code from above';"```

File: src/copyCodeBlock.js [hunk 3]
```diff
@@ -2,31 +2,57 @@ import { getDisplayString, getClipboardString } from './utils';
 import styles from './styles';
 
 export default (string, opts = {}) => {
-    const lang = opts && opts.lang || '';
+    const { lang = 'HTML', shouldReturnDomEl = false } = opts;
```
Comment (kgroat) [inline src/copyCodeBlock.js]: We shouldn't be defaulting `lang`:
* It'll always produce a warning if `hightlight.js` isn't installed
* We shouldn't try to syntax highlight if no language is supplied

File: src/copyCodeBlock.js [hunk 4]
```diff
@@ -2,31 +2,57 @@ import { getDisplayString, getClipboardString } from './utils';
...
+        textarea.select();
+
+        button.innerHTML = 'Copied!';
+        setTimeout(function() { button.innerHTML = '${buttonText}'; }, 1500);
+
+        document.execCommand('copy');
+        document.body.removeChild(textarea);
+    })();`;
+
+    if (shouldReturnDomEl) {
```
Comment (kgroat) [inline src/copyCodeBlock.js]: We don't have to have another code path here that constructs the DOM by hand; we can reuse the string below by doing:
```javascript
const domContent = `...stuff below...`;

const div = document.createElement('div');
div.innerHTML = domContent;
return div.children[0];

Comment (Pickra) [reply src/copyCodeBlock.js]: Ahh, tried that but forgot the `.children[0]`

File: src/copyCodeBlock.js (hunk 3 above)
Comment (Pickra) [reply src/copyCodeBlock.js]: `lang` is only being used for the `aria-label`, it's not mutating `opts.lang`.

That being said, you're still right. I don't want the screen reader to say "undefined code block"

**Comment by kgroat** [comment]
👍

**Pull request merged**